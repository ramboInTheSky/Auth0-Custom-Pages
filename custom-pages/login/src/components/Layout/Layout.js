import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router'
import Header from '../Header'
import Nav from '../Navigation'
import './Layout.scss'

export const CoreLayout = ({ children }) => (
  <div className="wrapper">
    <Header className="col leftCol" />
    <div className="mainContainer col rightCol">
      <Nav />
      {children}
      <Link className="forgotPasswordLink" data-qa-sel="forgotPassword-link" to="/forgotPassword">
        Forgot password?
      </Link>
    </div>
  </div>
)

CoreLayout.propTypes = {
  children: PropTypes.element.isRequired
}

export default CoreLayout
