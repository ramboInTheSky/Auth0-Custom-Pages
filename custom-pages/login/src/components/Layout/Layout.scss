@import "../variables.scss";

body {
  font-family: var(--font-base);
}

.wrapper {
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.col {
  height: 100%;
  padding: 3rem;
  width: var(--col-width-desktop);
}

.rightCol {
  padding-top: 1.4rem !important;
}

.mainContainer {
  padding-top: var(--top);
  min-width: var(--col-width-desktop);
}

@media (max-width: 700px) {
  .wrapper {
    flex-direction: column;
  }
  .rightCol,
  .leftCol {
    width: 100%;
    min-width: var(--col-width-mobile);
    padding-bottom: 0;
  }
  .mainContainer {
    min-width: var(--col-width-mobile);
  }

  .navigation {
    width: calc(var(--col-width-mobile)-5rem);
  }
}
