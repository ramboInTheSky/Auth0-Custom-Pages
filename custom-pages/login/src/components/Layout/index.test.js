import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-14'

import Layout from './'

Enzyme.configure({ adapter: new Adapter() })
const props = { children: <div /> }

describe('Layout components', () => {
  it('should render <Layout /> correctly', () => {
    expect(shallow(<Layout {...props} />)).toMatchSnapshot()
  })
})
