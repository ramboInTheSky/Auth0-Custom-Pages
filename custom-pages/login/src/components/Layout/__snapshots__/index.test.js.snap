// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout components should render <Layout /> correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CoreLayout>
    <div />
  </CoreLayout>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          className="col leftCol"
        />,
        <div
          className="mainContainer col rightCol"
        >
          <Navigation />
          <div />
          <Link
            className="forgotPasswordLink"
            data-qa-sel="forgotPassword-link"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/forgotPassword"
          >
            Forgot password?
          </Link>
        </div>,
      ],
      "className": "wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "col leftCol",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Navigation />,
            <div />,
            <Link
              className="forgotPasswordLink"
              data-qa-sel="forgotPassword-link"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/forgotPassword"
            >
              Forgot password?
            </Link>,
          ],
          "className": "mainContainer col rightCol",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Forgot password?",
              "className": "forgotPasswordLink",
              "data-qa-sel": "forgotPassword-link",
              "onlyActiveOnIndex": false,
              "style": Object {},
              "to": "/forgotPassword",
            },
            "ref": null,
            "rendered": "Forgot password?",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            className="col leftCol"
          />,
          <div
            className="mainContainer col rightCol"
          >
            <Navigation />
            <div />
            <Link
              className="forgotPasswordLink"
              data-qa-sel="forgotPassword-link"
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/forgotPassword"
            >
              Forgot password?
            </Link>
          </div>,
        ],
        "className": "wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "col leftCol",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Navigation />,
              <div />,
              <Link
                className="forgotPasswordLink"
                data-qa-sel="forgotPassword-link"
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/forgotPassword"
              >
                Forgot password?
              </Link>,
            ],
            "className": "mainContainer col rightCol",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Forgot password?",
                "className": "forgotPasswordLink",
                "data-qa-sel": "forgotPassword-link",
                "onlyActiveOnIndex": false,
                "style": Object {},
                "to": "/forgotPassword",
              },
              "ref": null,
              "rendered": "Forgot password?",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFourteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
